docs/source/user_guide/connector/connector_concurrency.rst

```diff
@@ -0,0 +1,95 @@
+==================================================
+
+.. toctree::
+   :maxdepth: 2
+
+Overview
+========
+Connector implements a mechanism to accelerate the data request process through a parallelism-based implementation. This feature allows users to retrieve data, in parallel, in less time.
+
+Concurrency in context: Youtube
+================================================================
+YouTube API provides several engaging endpoints, for example, the "videos" endpoint, which returns a collection of relevant videos matching a specified query.
+To start, let's review the existing configuration file for the "videos" endpoint available `here <https://github.com/sfu-db/DataConnectorConfigs/blob/develop/youtube/videos.json>`_::
+
+    {
+    "version": 1,
+    "request": {
+        "url": "https://www.googleapis.com/youtube/v3/search",
+        "method": "GET",
+        "authorization": {
+            "type": "QueryParam",
+            "keyParam": "key"
+        },
+        "params": {
+            "q": true,
+            "part": true,
+            "type": false,
+            "maxResults": false,
+            "pageToken": false
+        },
+        ...
+    }
+
+
+As you can see in the "params" property, five params are defined for this endpoint: ``q`` and ``part`` are mandatory, and the remaining ones are optional: ``type``, ``maxResults`` and ``pageToken``. See more details on the YouTube API specification webpage.
+Employing the ``q`` parameter, we can put any string to search for. The ``part`` parameter indicates the resource that the API response will include. It should always be set to ``snippet``.
+
+Now, we can execute our query::
+
+    import asyncio
+    from dataprep.connector import connect
+    youtube_connector = connect("youtube", _auth={"access_token":"<Your YouTube access token>"})
+    df = await youtube_connector.query("videos", q="data science", part="snippet")
+
+Let's review the obtained results:
+
+.. image:: ../../_static/images/tutorial/connector_youtube_query.png
+	:align: center
+   	:width: 964
+   	:height: 181
+
+For the "videos" endpoint, by default, the maximum number of search results to return for each request is 5. However, with Connector, you can also specify the desired record number you want to get by including the ``_count`` argument.
+
+Now, let's add the ``_count`` argument to the ``query`` function, and give it “300” as a value::
+
+    import asyncio
+    from dataprep.connector import connect
+    youtube_connector = connect("youtube", _auth={"access_token":"<Your YouTube access token>"})
+    df = await youtube_connector.query("videos", q="data science", part="snippet", _count=300)
+
+
+Let's execute this query again:
+
+.. image:: ../../_static/images/tutorial/connector_youtube_query_2.png
+	:align: center
+   	:width: 952
+   	:height: 367
+
+As you can see, we get exactly 300 records! The pagination feature enables us to get any amount of data as needed without learning the API detail.
```

Comment:
why introduce the auto-pagination feature in the concurrency block?

Reply:
It is included because YouTube endpoint limits the number of results to 5, so it is not possible to show the advantages of the concurrency feature with such a small number of results. Descriptions adjusted.